### PR TITLE
[CNXC-195] Standard Footer for Relevant Info

### PR DESCRIPTION
**Pull request by colinywu**
Added a consistently styled footer to the log-in and home page of the app, fixed href issues with previous code, and enabled routing from the logo/titles.

Implemented the core features and designs of the footer that displays the required information, with appropriate responsiveness and consistency with past designs/stylings

### CODE REVIEW

**Comment by boris-nguyen** [comment]
@colinywu can you confirm whether this PR is ready or work still needs to be done? I remember Audrey mentioning some minor changes? If changes are still needed, just so we don't merge it accidentally, can you convert this PR to draft? The `convert to draft` button is under the reviewers section. Thanks

File: src/pages/LogIn/Login.tsx
Comment (boris-nguyen) [inline src/pages/LogIn/Login.tsx:22]: Hey, could you do me a favour and change the alternative text to `DarwinAI logo`? Thanks

File: src/pages/Dashboard/Dashboard.tsx
Comment (boris-nguyen) [inline src/pages/Dashboard/Dashboard.tsx:51]: Please use `kebab-case` for CSS class names.

File: src/assets/scss/components/dashboard.scss
Comment (boris-nguyen) [inline src/assets/scss/components/dashboard.scss:53]: Please use `kebab-case` for CSS class names.

Comment (boris-nguyen) [inline src/assets/scss/components/dashboard.scss:53]: Since the footer is displayed on all pages, do you think it would be more appropriate to put the footer CSS in a separate file or in `main.scss`? @janakitti what are your thoughts about this?

File: src/containers/Layout/Footer.tsx
Comment (boris-nguyen) [inline src/containers/Layout/Footer.tsx:10]: Import statements are misaligned

File: src/containers/Layout/Footer.tsx
Comment (boris-nguyen) [inline src/containers/Layout/Footer.tsx:46]: Please remove trailing whitespace

Comment (boris-nguyen) [inline src/containers/Layout/Footer.tsx:46]: Misaligned code, inconsistent indentation throughout the whole file.
Screenshot from 2021-03-03 23-39-59

File: src/containers/Layout/Footer.tsx
Comment (boris-nguyen) [inline src/containers/Layout/Footer.tsx:40]: I would suggest removing `<React.Fragment>` since there is no need for it here. Also, for readability, it's better to split this on multiple lines.

File: src/containers/Layout/Footer.tsx
Comment (boris-nguyen) [inline src/containers/Layout/Footer.tsx:16]: I would suggest removing `<React.Fragment>` since there is no need for it here.

File: src/containers/Layout/Footer.tsx
Comment (boris-nguyen) [inline src/containers/Layout/Footer.tsx:19]: Please use `kebab-case` for CSS class names.

File: src/containers/Layout/Footer.tsx
Comment (boris-nguyen) [inline src/containers/Layout/Footer.tsx:22]: I would suggest leaving the space outside of `<a>`, otherwise, it would underline it as a hyperlink as well.

```suggestion
                    <p className="footerText">An open-source neural network for disease detection created by&nbsp;
                    <a href="https://www.darwinai.com/" target="_blank" rel="noopener noreferrer">DarwinAI</a>,&nbsp;
                    <a href="https://www.redhat.com/en" target="_blank" rel="noopener noreferrer">RedHat</a>, and&nbsp;
                    <a href="https://www.childrenshospital.org/" target="_blank" rel="noopener noreferrer">Boston Children's Hospital</a></p>
```

File: src/containers/Layout/Footer.tsx
Comment (boris-nguyen) [inline src/containers/Layout/Footer.tsx:26]: Please use `kebab-case` for CSS class names.

File: src/assets/scss/layout/layout.scss
Comment (boris-nguyen) [inline src/assets/scss/layout/layout.scss:34]: Please, nest these classes and ensure they all use `kebab-case`

**Review by boris-nguyen — CHANGES_REQUESTED** [review]
@colinywu please, take a look at the comments and resolve them in order for the code quality to improve. 

The requested changes are minor. 
Some of them are
- Use `<React.Fragment>` only if a function is returning multiple elements. Fragments are used to group multiple elements without adding extra HTML nodes in the DOM tree.
- Use `kebab-case` for SCSS class names
- Nest SCSS classes to avoid future SCSS conflicts as well as in order to have a clearer visual hierarchy
- Fix inconsistent indentation. In some places you use 2 spaces, in other 4 spaces. As well as, there are places where indentation was not needed. Please, ensure your VSCode has indentation set up properly. Hopefully, once linting and prettier are integrated, these issues would be avoided.
- Use `&nbsp;` for whitespace instead of including space at the beginning/end of a hyperlink

File: src/pages/LogIn/Login.tsx
Comment (colinywu) [reply src/pages/LogIn/Login.tsx]: Done!

File: src/assets/scss/components/dashboard.scss
Comment (colinywu) [reply src/assets/scss/components/dashboard.scss]: My only concern here is that changing to kebab-case isn't consistent with what's been done with the project before. Seems that previous work was done in a mix of camel-case and kebab-case. I've changed my work to use kebab-case, but I think it would also be worth it to go back and make naming conventions consistent, when there's time

Comment (boris-nguyen) [reply src/assets/scss/components/dashboard.scss]: I think the whole codebase needs to be refactored. There are lots of inconsistencies everywhere. I saw camelCase, PascalCase, kebab-case, snake_cake in the project. Depending on the tech stack, for certain things naming conventions are different depending on that what technology is being used. However, for CSS, no matter what technology, the standard is to either use kebab-case or BEM. So I firmly believe we should start making new code follow the right conventions so it takes less time to refactor later on.

Comment (colinywu) [reply src/assets/scss/components/dashboard.scss]: agreed, moved it into main.scss

File: src/assets/scss/layout/layout.scss
Comment (janakitti) [inline src/assets/scss/layout/layout.scss:39]: Should we be using `em`s instead of `px` for more responsive sizing?

File: src/containers/Layout/Footer.tsx
Comment (janakitti) [inline src/containers/Layout/Footer.tsx:10]: Use kebab-case for image file names for consistency.

File: src/containers/Layout/Footer.tsx
Comment (janakitti) [inline src/containers/Layout/Footer.tsx:27]:
```suggestion
          <p className="footer-text">Find the COVID-Net project&nbsp;<a href="https://github.com/darwinai" target="_blank" rel="noopener noreferrer">here:</a></p>
```

File: src/containers/Layout/Footer.tsx
Comment (janakitti) [inline src/containers/Layout/Footer.tsx:42]: Same comment about `em`s. May require changing the units for the header as well.

File: src/pages/Dashboard/Dashboard.tsx
Comment (janakitti) [inline src/pages/Dashboard/Dashboard.tsx:48]:
```suggestion
        <PageSection className="section-area" variant={PageSectionVariants.light}>
```

File: src/containers/Layout/Footer.tsx
Comment (janakitti) [inline src/containers/Layout/Footer.tsx:9]: Just something to consider: I noticed that the shade of blue used for the GitHub logo isn't the same as the primary blue colour used by PatternFly (`#06c`). Although, I'm not sure how strict we need to be with our colour scheme since it seems our `<a>` elements all use different shades of blue.

**Review by janakitti — COMMENTED** [review]
Nice work on designing the footer, @colinywu! Just added a few minor comments about style consistency.

File: src/pages/Dashboard/Dashboard.tsx
Comment (boris-nguyen) [reply src/pages/Dashboard/Dashboard.tsx:48]: Great suggestion! The practice for attribute values is indeed to use double quote marks.

Let's try to stick to double quote marks for new code and we can refactor the old code to have double quote marks later.

File: src/assets/scss/layout/layout.scss
Comment (boris-nguyen) [inline src/assets/scss/layout/layout.scss:33]: I might be wrong but it looks like that bracket is not needed there.

File: src/assets/scss/layout/_layout.scss
Comment (boris-nguyen) [inline src/assets/scss/layout/_layout.scss:14]: Indentation is inconsistent, please ensure the indentation is 2 spaces.

File: src/containers/Layout/Footer.tsx
Comment (colinywu) [reply src/containers/Layout/Footer.tsx]: Yeah, i agree. the colour for the github blue isn't a perfect match, but it's nearly impossible to find an appropriate logo/icon that matches the colour scheme perfectly. I think that it looks extremely close enough to have the UI consistent.

And yeah, I also noticed that the shades of blue for the anchors are a little different as well, I'd say this is an issue to address later when refactoring the UI

File: src/assets/scss/layout/layout.scss
Comment (colinywu) [reply src/assets/scss/layout/layout.scss:33]: that bracket is needed for footer-link

footer-logo is nested inside footer-link., so there has to be 2 sets of braces for them

File: src/assets/scss/layout/_layout.scss
Comment (boris-nguyen) [inline src/assets/scss/layout/_layout.scss]: Whenever there are nested selectors, put nested selectors at the end. Ensure to leave a whitespace between your rule declaration and nested selectors.

File: src/assets/scss/layout/layout.scss
Comment (boris-nguyen) [reply src/assets/scss/layout/layout.scss:33]: Makes sense, can you please properly indent `.footer-logo` so it looks like it's actually nested? Thanks

File: src/assets/scss/layout/layout.scss
Comment (colinywu) [reply src/assets/scss/layout/layout.scss:39]: Changed much of the px units to em, but as discussed in call, much of the project is still using px measurements and struggles with responsiveness, so it's best to have proper discussion about how to make COVID-Net more responsive and what units would be best for the project and do a proper overhaul of legacy code

File: src/containers/Layout/Footer.tsx
Comment (colinywu) [reply src/containers/Layout/Footer.tsx:40]: Don't think changing this would be the best call, since the previous header used 300 x 190px measurements. I followed convention to make it more consistent. I do agree that the px measurements are not dynamic, but I think it would require further testing and proper discussion to make the UI flexible and consistent

File: src/containers/Layout/Header.tsx
Comment (boris-nguyen) [inline src/containers/Layout/Header.tsx:83]: In my personal opinion, I'd not use inline TS code for the onClick events. Leave this as it is for now. However, as a team, we might want to discuss later in the future what would be the best practices for this project such as where inline TS code is acceptable.

File: src/containers/Layout/Footer.tsx
Comment (boris-nguyen) [inline src/containers/Layout/Footer.tsx:7]:
```suggestion
import { useHistory } from "react-router-dom";
```

File: src/assets/scss/layout/layout.scss
Comment (colinywu) [reply src/assets/scss/layout/layout.scss:33]: Done!

File: src/containers/Layout/Footer.tsx
Comment (boris-nguyen) [inline src/containers/Layout/Footer.tsx:39]:
```suggestion
      onClick={(e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => { history.push("/"); e.preventDefault() }}
```

File: src/containers/Layout/Footer.tsx
Comment (boris-nguyen) [inline src/containers/Layout/Footer.tsx:40]:
```suggestion
      href="/#">
        <img src={logo} className="logo" alt="DarwinAI Logo" height="300px" width="190px" />
```

File: src/containers/Layout/Header.tsx
Comment (colinywu) [reply src/containers/Layout/Header.tsx:83]: Understood, I went with the quick inline fix since it was the shortest solution to fix the issue with the anchors. Ideally, we should be using buttons instead of anchors here, as we discussed earlier

File: src/containers/Layout/Footer.tsx
Comment (boris-nguyen) [inline src/containers/Layout/Footer.tsx:40]: If time permits, please, create a class for DarwinAI logo with height and width rule declarations. I noticed that both Footer and Header have those same attributes. This would help us with re-usability. In addition, "separation of concerns" concept helps with future debugging.

Comment (colinywu) [reply src/containers/Layout/Footer.tsx]: Great suggestion, will do that right away

File: src/containers/Layout/Footer.tsx
Comment (boris-nguyen) [inline src/containers/Layout/Footer.tsx:28]: If time permits, can you change the alternative text to something along the lines of "DarwinAI COVID-Net UI Github link" or another text that you feel is more descriptive and more appropriate?

File: src/containers/Layout/Header.tsx
Comment (boris-nguyen) [reply src/containers/Layout/Header.tsx:83]: Resolving the comment for now, something to discuss later on as a team.

File: src/containers/Layout/Footer.tsx
Comment (janakitti) [inline src/containers/Layout/Footer.tsx:42]: It appears that PatternFly's PageHeader Logo is implemented as an `<a>`. This means that by passing our logo `img` wrapped in an anchor, have an `<a>` as a descendant of an `<a>`, which is not allowed.

We could just have the `onClick` as an attribute directly on the `img` itself and drop the anchor wrapper:
```
<img onClick={() => { history.push("/") }} src={logo} className="logo" alt="DarwinAI Logo" />)}>
```

The same change would need to be made to the Header.

Comment (colinywu) [reply src/containers/Layout/Footer.tsx:41]: Awesome suggestion, cleans up the solution a ton

File: src/assets/scss/layout/layout.scss
Comment (janakitti) [inline src/assets/scss/layout/layout.scss:50]: Might need to specify a `min-height` here.
Screenshot from 2021-03-08 14-26-11

File: src/assets/scss/layout/layout.scss
Comment (janakitti) [inline src/assets/scss/layout/layout.scss:13]: Don't remember what the final decision was for the positioning of the footer, but if we wanted the footer to stick to the bottom of the viewport and not overlap the contents of the page, we can set `position: absolute` here and set the `height: 100vh` for the Page component inside PageWrapper.

Comment (colinywu) [reply src/assets/scss/layout/layout.scss:13]: Is there any reason to use position `absolute` over `fixed`?

File: src/containers/Layout/Footer.tsx
Comment (janakitti) [inline src/containers/Layout/Footer.tsx:20]:
```suggestion
            <a href="https://chrisproject.org/" target="_blank" rel="noopener noreferrer">ChRIS</a>, and&nbsp;
            <a href="https://redhat.com/chris" target="_blank" rel="noopener noreferrer">RedHat</a>&nbsp;</p>
```

File: src/containers/Layout/Header.tsx
Comment (janakitti) [inline src/containers/Layout/Header.tsx:28]:
```suggestion
  const [isDropdownOpen, setIsDropdownOpen] = React.useState(false);
```

File: src/pages/Dashboard/Dashboard.tsx
Comment (janakitti) [inline src/pages/Dashboard/Dashboard.tsx:26]:
```suggestion
        });
```

File: src/pages/Dashboard/Dashboard.tsx
Comment (janakitti) [inline src/pages/Dashboard/Dashboard.tsx:30]:
```suggestion
        });
```

File: src/assets/scss/layout/layout.scss
Comment (janakitti) [inline src/assets/scss/layout/layout.scss:49]:
```suggestion
  min-width: 190px;
```

**Review by janakitti — CHANGES_REQUESTED** [review]

File: src/pages/Dashboard/Dashboard.tsx
Comment (janakitti) [inline src/pages/Dashboard/Dashboard.tsx:17]:
```suggestion
    document.title = "Analysis - COVID-Net UI";
```

**Review by janakitti — APPROVED** [review]
Awesome work on the footer, @colinywu!

**Review by boris-nguyen — APPROVED** [review]

**Review by IvanToronto — APPROVED** [review]

**Pull request merged**
